qa/python_models/response_sender/model_common.py

```diff
@@ -0,0 +1,185 @@
+# Copyright 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+#  * Redistributions of source code must retain the above copyright
+#    notice, this list of conditions and the following disclaimer.
+#  * Redistributions in binary form must reproduce the above copyright
+#    notice, this list of conditions and the following disclaimer in the
+#    documentation and/or other materials provided with the distribution.
+#  * Neither the name of NVIDIA CORPORATION nor the names of its
+#    contributors may be used to endorse or promote products derived
+#    from this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+# EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+# PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import asyncio
+import threading
+import time
+
+import numpy as np
+
+
+class ResponseSenderModelCommon:
+    def __init__(self, pb_utils):
+        self._pb_utils = pb_utils
+        self._background_tasks = set()
+
+    def _get_instructions_from_request(self, request):
```

Comment:
```suggestion
    def _get_instructions_from_request(self, request):
     """Determine the execution instructions from the inputs. This test tries
    to examine all the corner cases with using response sender. It uses the
    following inputs to control the model behaviour:
    
    Assumptions: The request batch size can be larger than one.
    
    There are 5 inputs in the model that control the model behaviour:
    
    * NUMBER_OF_RESPONSE_BEFORE_RETURN(UINT8): Determines the number of responses before returning from execute function.
    * SEND_COMPLETE_FINAL_FLAG_BEFORE_RETURN(BOOL): Determines whether the final flag will be sent before return.
    * RETURN_A_RESPONSE(BOOL): Return the response when the model is returning from `execute` function.
    * NUMBER_OF_RESPONSE_AFTER_RETURN(UINT8): Determines the number of responses after return.
    * SEND_COMPLETE_FINAL_FLAG_AFTER_RETURN(BOOL): Determines whether the final flag will be sent before return.
    
    Note that if the batch size is larger than one the sum of the values in the batch size will be used for determining the value of each input. The response_id is used to determine the difference between responses
    sent before the execute vs the responses sent after the execute.
    """

```

Reply:
Nice summary! Clarified some statements based on my understanding and added to this PR.